src/Common/src/System/Collections/Generic/LargeArrayBuilder.cs

```diff
@@ -216,25 +216,36 @@ public CopyPosition CopyTo(CopyPosition position, T[] array, int arrayIndex, int
             int row = position.Row;
             int column = position.Column;
 
-            for (; count > 0; row++, column = 0)
+            T[] buffer = GetBuffer(row);
+            int copied = CopyToCore(buffer, column);
+
+            while (count > 0)
+            {
+                buffer = GetBuffer(++row);
+                copied = CopyToCore(buffer, 0);
+            }
+
+            return copied == buffer.Length
+                ? new CopyPosition(row + 1, 0)
+                : new CopyPosition(row, copied);
+
+            int CopyToCore(T[] sourceBuffer, int sourceIndex)
```

Comment:
Nit: Is it customary in corefx to put a local func at the very end of the containing method?

Reply:
We have very few local methods; there's no convention yet.  But unless there's an obviously better place, end of the method makes sense.

Reply:
I'm not sure what is the custom for this repo. I've not seen a lot of local functions used here yet. But that pattern is common.